addons/content-index/src/main/java/org/commonjava/indy/content/index/conf/HostedContentIndexRescanManager.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.content.index.conf;
+
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.change.event.ArtifactStorePostRescanEvent;
+import org.commonjava.indy.change.event.ArtifactStorePreRescanEvent;
+import org.commonjava.indy.content.ContentManager;
+import org.commonjava.indy.content.DownloadManager;
+import org.commonjava.indy.content.StoreResource;
+import org.commonjava.indy.content.index.ContentIndexManager;
+import org.commonjava.indy.content.index.IndexedStorePath;
+import org.commonjava.indy.data.IndyDataException;
+import org.commonjava.indy.data.StoreDataManager;
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.Group;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This component will handle the content index for the hosted repo during its rescan. It will rebuild all relevant
+ * index entries for the hosted repo, and remove all useless entries for the missed artifacts in hosted repo
+ */
+@ApplicationScoped
+public class HostedContentIndexRescanManager
```

Comment:
We should make an interface like ContentIndexRescanManager and make this implement it. Then, we can make the Koji one implement and add any new add-on specific behaviors by simply implementing the interface. We can iterate the rescan managers available in the deployed system and use them to execute their specific rescanning actions. We would need to make the method names generic like `preRescan` and `postRescan` or similar...